hedera-node/test-clients/build.gradle.kts

```
@@ -230,6 +232,21 @@ val yahCliJar =
         }
     }
 
+val rcdiffJar =
+    tasks.register<ShadowJar>("rcdiffJar") {
+        exclude(listOf("META-INF/*.DSA", "META-INF/*.RSA", "META-INF/*.SF", "META-INF/INDEX.LIST"))
+        from(sourceSets["rcdiff"].output)
```

@jjohannes Is this the correct way to reference the source set?

---

Note that we configure all _shadowJar_ tasks are configured to have `from(sourceSets.main.output)` as default. If this should be replaced in his case, you should use `setFrom()` (instead of `from()`).

```suggestion
        setFrom(sourceSets["rcdiff"].output)
```

---

`setFrom()` doesn't seem to exist.

---

Sorry my bad. I mixed that up with other APIs. So I think we can't override it IIRC.
Maybe it is okay for this case to leave it like this?
Otherwise we would need to do adjustments in `com.hedera.hashgraph.shadow-jar.gradle.kts` and then check every of the other `ShadowJar` tasks. Maybe do this in a follow up, if we care.

---

I'll leave it for a follow-up, this is already past the limits of my Gradle knowledge. 😁